casper/src/test/scala/coop/rchain/casper/api/CreateBlockAPITest.scala

```diff
@@ -108,7 +108,7 @@ private class SleepingMultiParentCasperImpl[F[_]: Monad: Time](underlying: Multi
       b: BlockMessage,
       handleDoppelganger: (BlockMessage, Validator) => F[Unit]
   ): F[BlockStatus]                                           = underlying.addBlock(b, ignoreDoppelgangerCheck[F])
-  def contains(b: BlockMessage): F[Boolean]                   = underlying.contains(b)
```

Comment:
Why this change? Please provide an overload that allows for passing the block itself too and delegates to the BlockHash-taking one.

Please also restore the previous usages of the block-taking method - I find the change from `.contains(block)` to `.contains(block.blockHash)` is downgrade in readability and a needless change.

Reply:
LOL sorry it occurred to me only under the shower that you're making a production code change (which is most likely justified - didin't notice it previously) - and I'm referring to test code.

I think what should happen is: we get a HashSetCasperTestNode#contains(BlockMessage) method that does the convenience of grabbing the hash and delegates to `casperEff.contains(hash)`.

So a change to all the call-sites will be needed anyway. I'd just prefer the change to be from:
`node.casperEff.contains(block)`
to
`node.contains(block)`
vs
`node.casperEff.contains(block.blockHash)`

Reply:
Well, the change was motivated by the fact that Casper only needs `hash` to check if the block exists or not. Providing the whole `BlockMessage` as an argument only to check if there is a block with given hash was bad design. Simple as that.